### PR TITLE
Filter out interactions from the panel based on their visibility

### PR DESCRIPTION
**Pull request by krampstudio**
Related https://oat-sa.atlassian.net/browse/AUT-1980

 - check if each interaction is visible in the authoring using the key pattern `taoQtiItem/creator/interaction/<interaction>`
 - same for PCI using `taoQtiItem/creator/customInteraction/<typeIdentifier>`
 - A small util has been added to check if a qti element is visible (and cleanup the helper)

### How to test:
 - run the unit test : `cd tao/views/build && npm ci && npx grunt connect:test taoqtiitemtest`
 - configure some interactions to be hidden in `config/tao/client_lib_config_registry.conf.php`, for example
```
                'taoQtiItem/creator/interaction/endAttempt' => 'hide',
                'taoQtiItem/creator/customInteraction/likertScale' => 'hide',
``` 
  and open the authoring to check if the interactions aren't available in the sidebar.

### CODE REVIEW

File: views/js/qtiCreator/editor/interactionsPanel.js [hunk 1]
```diff
@@ -23,28 +23,27 @@
...
     'use strict';
 
     /**
      * Set up the interaction selection panel
      * @param {jQueryElement} $container - the panel container
      */
-    return function setUpInteractionPanel($container){
+    return function setUpInteractionPanel($container) {
+        const interactions = qtiElements.getAvailableAuthoringElements();
+        //const availableInteractions = qtiElements.getAvailableAuthoringElements();
```
Comment (yukipastelcat) [inline views/js/qtiCreator/editor/interactionsPanel.js:39]: Do we still need this comment?

**Review by yukipastelcat — APPROVED** [review]
Feature works good, good job👍

**Review by shaveko — COMMENTED** [review]
Feature works well, but when I run unit tests, I've got these warnings/erorrs
Screenshot 2022-04-21 at 11 05 12

File: views/js/portableElementRegistry/factory/ciRegistry.js [hunk 2]
```diff
@@ -30,31 +30,46 @@ define([
...
                     return {
-                        label : pciModel.label, //currently no translation available
-                        icon : pciModel.creator.icon.replace(new RegExp('^' + typeIdentifier + '\/'), pciModel.baseUrl),
-                        short : pciModel.short,
-                        description : pciModel.description,
-                        qtiClass : 'customInteraction.' + pciModel.typeIdentifier, //custom interaction is block type
-                        tags : _.union([__('Custom Interactions')], pciModel.tags),
-                        group : 'custom-interactions'
+                        label: pciModel.label, //currently no translation available
+                        icon: pciModel.creator.icon.replace(new RegExp('^' + typeIdentifier + '/'), pciModel.baseUrl),
```
Comment (shaveko) [inline views/js/portableElementRegistry/factory/ciRegistry.js:56]: Eslint suggests to use template instead of string concatenation here

**Review by shaveko — APPROVED** [review]
Works well, looks nice - 
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**